### PR TITLE
Prevent annoying error in VSmac

### PR DESCRIPTION
**Pull request by sandyarmstrong**
Work around https://github.com/mono/monodevelop/issues/3859, which causes an annoying popup when opening certain SDK-style projects that lack a `TargetFramework` in VSmac.

### CODE REVIEW

File: test/Microsoft.ML.Tests/Microsoft.ML.Tests.csproj [hunk 1]
```diff
@@ -1,4 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+  </PropertyGroup>
```
Comment (Ivanidzo4ka) [inline test/Microsoft.ML.Tests/Microsoft.ML.Tests.csproj]: Should we add this to all our tests projects?

Comment (sandyarmstrong) [reply test/Microsoft.ML.Tests/Microsoft.ML.Tests.csproj]: I honestly don't know. I was just doing the minimal change to work around the VSmac bug. I'm not sure why it's not complaining about the other projects.

Comment (Ivanidzo4ka) [reply test/Microsoft.ML.Tests/Microsoft.ML.Tests.csproj]: https://github.com/dotnet/machinelearning/blob/master/test/Directory.Build.props
we have it specified for whole directory. And it works on Windows, I wonder is it VS mac specific

Comment (Ivanidzo4ka) [reply test/Microsoft.ML.Tests/Microsoft.ML.Tests.csproj]: Let me add @eerhardt  as most experienced person with builds and projects

Comment (sandyarmstrong) [reply test/Microsoft.ML.Tests/Microsoft.ML.Tests.csproj]: Yes, the bug is 100% VSmac-specific. I hadn't noticed the props file but you're definitely right.

Up to you folks if you want to carry a work-around or just tell users to ignore the error until VSmac gets fixed.

Comment (eerhardt) [reply test/Microsoft.ML.Tests/Microsoft.ML.Tests.csproj]: @mhutch @mrward - any idea how long until the VSmac bug is fixed?  If it should be fixed soon (like in the next release) maybe we shouldn't take this.  But if it is going to be a while, we probably should make it so VSmac works with our repo.

Comment (eerhardt) [reply test/Microsoft.ML.Tests/Microsoft.ML.Tests.csproj]: If we go ahead with this change, I agree that all the test projects should be fixed.  We should probably just take the TargetFramework setting out of the test/Directory.Build.props file, and duplicate them in all the test .csprojs.

Comment (mrward) [reply test/Microsoft.ML.Tests/Microsoft.ML.Tests.csproj]: It is currently targeted for VS Mac 7.6, which would be the next release, but there is no fix as yet.

Comment (sandyarmstrong) [reply test/Microsoft.ML.Tests/Microsoft.ML.Tests.csproj]: @eerhardt I've updated the PR to move `TargetFramework` out of the props file and into each test csproj.

**Pull request merged**